### ISSUE TITLE
Fail to Build documentation in Xcode 14 beta 5

### ISSUE THREAD

**Issue by Kyle-Ye** [vapor/vapor#2879]
### Describe the bug

Build documentation in Xcode 14 beta 5 will get the following error
```shell
Symbol with identifier 's:So9NSDecimala10FoundationE11FormatStyleV7PercentV' couldn't be found
```

See the full log
[Build Documentation vapor-Package_2022-08-27T00-22-18.txt](https://github.com/vapor/vapor/files/9434800/Build.Documentation.vapor-Package_2022-08-27T00-22-18.txt)

### To Reproduce

1. Clone the repo with main (commit ID: 45bc9ab1e3c4ba309606762664fc18358b7c92e5)
2. Use Xcode 14 beta 5 to build documentation (Open Package.swift and press ⌃⇧⌘D)

### Expected behavior

Build documentation successfully.

### Environment

* OS version: macOS 13.0 Beta(22A5331f)

### Additional context

This will also make the client's documentation building to fail if the client depends on vapor.

After some investigation I found that this strange symbol is from a strange symbol graph file - `Vapor@_StringProcessing.symbols.json`

```
.
├── Vapor.symbols.json
├── Vapor@CoreFoundation.symbols.json
├── Vapor@Dispatch.symbols.json
├── Vapor@Foundation.symbols.json
├── Vapor@NIOCore.symbols.json
├── Vapor@NIOHTTP1.symbols.json
├── Vapor@ObjectiveC.symbols.json
├── Vapor@Swift.symbols.json
├── Vapor@_Concurrency.symbols.json
└── Vapor@_StringProcessing.symbols.json (*)

~/Library/Developer/Xcode/DerivedData/vapor-dkxjpoxzmdidwzaizjdacefibuph/Build/Intermediates.noindex/vapor.build/Debug/Vapor.build/symbol-graph/swift/arm64-apple-macos/Vapor@_StringProcessing.symbols.json
```

**Comment by Kyle-Ye** [comment]
Fixed in main/Swift 5.8 and cherry-picked to Swift 5.7.

**Comment by Kyle-Ye** [comment]
Fixed by https://github.com/apple/swift/pull/61049
